### DISCUSSION TITLE
Creating new worktree for a local repository?

### DISCUSSION THREAD

**Discussion by NobodyXu** [GitoxideLabs/gitoxide#1002] [Q&A] [answered]
Is it possible to create a new worktree for a local repository?
I'd like to use it as an optimization in `cargo-binstall`.

There is indeed a `Worktree` but I'm not sure whether it supports this functionality.

**Comment by Byron** [comment, marked as answer]
The `Worktree` doesn't have any capability to mutate itself yet, nor is there a way to create an official worktree like `git worktree add` does. However, creating worktree checkouts is *relatively* straightforward with [gix::worktree::state::checkout()](https://docs.rs/gix-worktree-state/0.1.0/gix_worktree_state/fn.checkout.html).  Further, you could create a [worktree_stream](https://docs.rs/gix/0.52.0/gix/struct.Repository.html#method.worktree_stream) which allows to examine the whole worktree (as it would be checked out) in memory in a streaming fashion.

Since `cargo-binstall` controls these directories, getting a plain checkout should be enough.

**Reply by NobodyXu** [reply]
Thank you!

Just a small question, can I simply pass `|_oid, data| BlobRef { data: &*data }` to parameter `find`?

**Reply by Byron** [reply]
I recommend to find references of the function in question - it should bring up multiple uses which can serve as example. Then it's something like `{ repo.objects.find_blob(oid, buf) }`.
